File: PushLocal/src/main/java/com/beariksonstudios/automatic/pushlocal/pushlocal/activities/main/DiscoveryTimer.java
package com.beariksonstudios.automatic.pushlocal.pushlocal.activities.main;

import android.content.Context;
import android.widget.Toast;
import com.beariksonstudios.automatic.pushlocal.pushlocal.activities.main.MainActivity;
import com.beariksonstudios.automatic.pushlocal.pushlocal.server.Device;

import java.util.ArrayList;
import java.util.TimerTask;

/**
 * Created by nphel on 9/26/2015.
 */
public class DiscoveryTimer extends TimerTask{

    private final ArrayList<Device> devices;
    private final MainActivity context;
    int numberDevices;

    public DiscoveryTimer(ArrayList<Device> devices, MainActivity context){
        this.devices = devices;
        this.context = context;
        numberDevices = devices.size();
    }
    @Override
    public void run() {
        context.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                Toast.makeText(context, devices.size() - numberDevices + " new devices found", Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: PushLocal/src/main/java/com/beariksonstudios/automatic/pushlocal/pushlocal/activities/main/DiscoveryTimer.kt
package com.beariksonstudios.automatic.pushlocal.pushlocal.activities.main

import android.widget.Toast
import com.beariksonstudios.automatic.pushlocal.pushlocal.server.Device

import java.util.ArrayList
import java.util.TimerTask

/**
 * Created by nphel on 9/26/2015.
 */
class DiscoveryTimer(private val devices: ArrayList<Device>, private val context: MainActivity) : TimerTask() {
    private var numberDevices: Int = 0

    init {
        numberDevices = devices.size
    }

    override fun run() {
        context.runOnUiThread {
            Toast.makeText(context, "${devices.size - numberDevices} new devices found", Toast.LENGTH_SHORT).show()
        }
    }
}
